tasks/main.yml

```diff
@@ -1 +1,155 @@
+- name: 4.2 Creating directories for templates and images
+  become: "{{ osp_director_become }}"
+  become_user: "{{ osp_director_become_user }}"
+  file:
+    path: "{{ item }}"
+    state: directory
+    mode: 0755
+  with_items:
+    - ~/images
+    - ~/templates
+  tags:
+    - "4.2"
+
+- name: 4.5 Installing the director packages
+  become: "{{ osp_director_become }}"
+  become_user: "{{ osp_director_become_user }}"
+  block:
+    - name: Install main Director packages
+      package:
+        name: "{{ osp_director_base_packages }}"
+        state: present
+
+    - name: install tempest-related packages
+      package:
+        name: "{{ osp_director_tempest_packages }}"
+        state: present
+      tags:
+        - tempest
+
+    - name: Add useful aliases for OpenStack
+      become: true
+      become_user: stack
+      blockinfile:
+        path: ~/.bashrc
+        block: |-
+          alias os="openstack"
+          alias osp="openstack"
+  tags:
+    - "4.5"
+
+- name: 4.6 Installing ceph-ansible
+  become: "{{ osp_director_become }}"
+  become_user: "{{ osp_director_become_user }}"
+  block:
+    # This is required if using ceph-ansible. Do not use the package that comes
+    # from @rhel-7-server-openstack-13-rpms
+    - name: 4.6 Enable the Ceph Tools repository
+      rhsm_repository:
+        name:
+          - rhel-7-server-rhceph-3-tools-rpm
+        state: enabled
+
+    - name: 4.6 Install ceph-ansible
+      package:
+        name: ceph-ansible
+        state: present
+  when: osp_director_undercloud.install_ceph_ansible
+
+- name: 4.9 Installing the director
+  become: true
+  become_user: stack
+  block:
+    - name: 4.9 Copy undercloud.conf to /home/stack
+      copy:
+        src: "{{ osp_director_templates }}/undercloud.conf"
+        dest: /home/stack/undercloud.conf
+        mode: preserve
+        remote_src: true
+
+    - name: 4.9 openstack undercloud install
+      command: openstack undercloud install
+      changed_when: false
+      register: _osp_install_undercloud
+      retries: 2
+      until: _osp_install_undercloud is success
+      async: 14400  # 4 hours
+      poll: 300  # every 5 minutes
+  tags:
+    - "4.9"
+
+- name: Source stackrc
+  shell: |
+    set -e
+    source ~/stackrc
+    env
+  register: _osp_director_stackrc_output
+  changed_when: false
+
+- name: Store stackrc for later
+  set_fact:
+    osp_director_os_env_data: >-
+      {{ _osp_director_stackrc_output.stdout
+      | env_to_dict(only=osp_director_env_filters) }}
+
+- name: 4.10 Obtaining images for overcloud nodes
+  become: true
+  become_user: stack
+  block:
+    - name: 4.10 Install rhosp-director-image packages
+      become: "{{ osp_director_become }}"
+      become_user: "{{ osp_director_become_user }}"
+      package:
+        name: "{{ osp_director_image_packages }}"
+        state: present
+
+    - name: 4.10 Make ~/images directory
+      file:
+        path: ~/images
```

Comment:
I read that there may be a better way to reference home dirs [here](https://github.com/oasis-roles/ocp_azure_driver/pull/1#discussion_r373098452).

Reply:
See previous comment.